frontend/src/lib/WorkflowParser.ts

```diff
@@ -376,7 +377,7 @@ export default class WorkflowParser {
     // Get the task name that corresponds to the nodeId
     let taskName = '';
     let taskStatus: NodePhase = NodePhase.SUCCEEDED;
-    for (const taskRunId of Object.getOwnPropertyNames(workflow.status.taskRuns || {})) {
+    for (const taskRunId of Object.getOwnPropertyNames(workflow.status.taskRuns)) {
```

Comment:
Custom task is not a taskRuns, I remember this will break the custom task node in the previous PR.